src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -110,14 +114,68 @@ private IntPtr AllocException(Exception ex)
         [DllImport(JitSupportLibrary)]
         private extern static char* GetExceptionMessage(IntPtr obj);
 
-        public static void Startup()
+
+        public static void Startup(TargetOS targetOS, TargetArchitecture targetArchitecture)
         {
-            jitStartup(GetJitHost(JitConfigProvider.Instance.UnmanagedInstance));
+            Debug.Assert(s_jitLibrary == IntPtr.Zero);
+
+            string crossgenPath = Path.GetDirectoryName(typeof(CorInfoImpl).Assembly.Location);
+            string jitTargetPath = Path.Combine(crossgenPath, GetTargetSpec(targetOS, targetArchitecture));
```

Comment:
We are renaming the JIT .dll anyway (from clrjit to clrjitilc). Would it be simpler to have the target arch in the name, so that we do not have to deal with loading it from subdirectories?

E.g. if the JIT dll was next to `typeof(CorInfoImpl).Assembly.Location`, you can just use the longer overload of `NativeLibrary.Load` to load it and it will take care of the prefixes and suffixes for you.

Reply:
I agree it's probably easier for general maintenance, will do. (With the downside that I'll need to fork the call to NativeLibrary.Load for the "default case" vs. "explicitly specified jitPath".)

Reply:
> fork the call to NativeLibrary.Load for the "default case" vs. "explicitly specified jitPath".

Sounds good.

Reply:
Fixed in 21st commit.